cmake/platforms/OpenCV-Darwin.cmake

```diff
@@ -1 +1,3 @@
-# empty
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "arm64.*|ARM64.*")
```

Comment:
I'm not sure if we should directly add "Darwin" for this purpose.
From the perspective of "Apple M1", yes, this is the straightway.
On the other hand, those instructions are available under complete Arm compatibility.
Thus, this is NOT Apple M1 specific.

We already have [CPU Dispatch](https://github.com/opencv/opencv/blob/master/cmake/OpenCVCompilerOptimizations.cmake) mode implemented, and I highly recommend leveraging this mode.

Here's my suggestion.

- [ ] Remove this file, at least for this PR purpose
- [ ] Add corresponding compiler flags in to https://github.com/opencv/opencv/blob/master/cmake/OpenCVCompilerOptimizations.cmake